tests/components/sensor/test_arlo.py

```diff
@@ -0,0 +1,185 @@
+"""The tests for the Netgear Arlo sensors."""
+import asyncio
+from collections import namedtuple
+import unittest
+from unittest.mock import patch, MagicMock
+from homeassistant.const import (
+    DEVICE_CLASS_TEMPERATURE, DEVICE_CLASS_HUMIDITY, ATTR_ATTRIBUTION)
+from homeassistant.components.sensor import arlo
+from homeassistant.components.arlo import DATA_ARLO
+from homeassistant.helpers import dispatcher
+from tests.common import get_test_home_assistant
+
+
+class TestArloSensor(unittest.TestCase):
+    """Test Netgear Arlo sensors."""
+
+    def _setup_arlo(self):
+        self.hass.data = {
+            DATA_ARLO: 'test'
+        }
+
+    @staticmethod
+    def _get_mock_sensor(name='Last', sensor_type='last_capture', data=None):
+        if data is None:
+            data = {}
+        return arlo.ArloSensor(name, data, sensor_type)
+
+    @staticmethod
+    def _get_named_tuple(input_dict):
+        return namedtuple('Struct', input_dict.keys())(*input_dict.values())
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """Setup shared dependencies for each test."""
+        self.hass = get_test_home_assistant()
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """Tear down shared dependencies for each test."""
+        self.hass.stop()
+
+    def test_setup_with_no_data(self):
+        """Test setup_platform with no data."""
+        result = arlo.setup_platform(self.hass, None, None)
+        self.assertEqual(result, False)
+
+    sensors = None
+
+    def _add_devices(self, sensors, boolean):
+        if boolean:
+            self.sensors = sensors
+
+    def test_setup_with_valid_data(self):
+        """Test setup_platform with valid data."""
+        config = {
+            'monitored_conditions': [
+                'last_capture',
+                'total_cameras',
+                'captured_today',
+                'battery_level',
+                'signal_strength',
+                'temperature',
+                'humidity',
+                'air_quality'
+            ]
+        }
+
+        self.hass.data[DATA_ARLO] = self._get_named_tuple({
+            'cameras': [self._get_named_tuple({
+                'name': 'Camera',
+                'model_id': 'ABC1000'
+            })],
+            'base_stations': [self._get_named_tuple({
+                'name': 'Base Station',
+                'model_id': 'ABC1000'
+            })]
+        })
+
+        arlo.setup_platform(self.hass, config, self._add_devices)
+        self.assertEqual(len(self.sensors), 8)
+
+    def test_sensor_name(self):
+        """Test the name property."""
+        sensor = TestArloSensor._get_mock_sensor()
```

Comment:
The method is weirdly named since it doesn't seem to return a mock sensor but an actual instance of an arlo sensor.

Access methods using `self.`.

Reply:
Will correct the method name and change to `self` access.

Also, just for reference as I'm coming back to python several years away, is it not optimal to use the `@staticmethod` decorator and treat these as "static" as opposed to members? Or is this just convention? Or is this an artifact of having this in a class as opposed to just standalone functions? Thanks for the help.

Reply:
Static methods can still be accessed on the instance, ie using `self.` inside instance methods. It's only inside the static method that the instance isn't available.

Yeah, the static methods are part of what could be cleaned up if we would move to standalone pytest tests.